bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/SearchHistoryMenu.java

```diff
@@ -93,6 +94,36 @@ public Control createContents(Composite parent) {
 	}
 
 	private void attachTableListeners() {
+		table.addKeyListener(new KeyListener() {
```

Comment:
You can use KeyAdapter to avoid empty method implementations

Reply:
I knew this comment might come but chose to stick to a regular "KeyListener" The reason is twofold:

1) I have state which I don't want to manage in the outer class (i.e. indexInTable). In my view, that is responsibility of the keylistener which manages where to scoll to
2) I also want to encapsulate `moveSelectionInTable`

What do you think?

Reply:
It makes sense to have both within the listener, so that's fine for me. But why wouldn't that be possible with a KeyAdapter? Only different should be that you can avoid empty method implementations, but you still have an anonymous class in which you can encapsulate the things you have mentioned.

Reply:
Fixed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/b6186f57fbec28d23260fc3a2f3d1101b910c3c5
see comment down below